view/frontend/web/js/view/payment/method-renderer/omise-offline-tesco-method.js

```diff
@@ -0,0 +1,140 @@
+        quote,
+        urlBuilder,
+        errorProcessor
+    ) {
+        'use strict';
+
+        return Component.extend({
+            defaults: {
+                template: 'Omise_Payment/payment/offline-tesco-form'
+            },
+
+            isPlaceOrderActionAllowed: ko.observable(quote.billingAddress() != null),
+
+            /**
+             * Get payment method code
+             *
+             * @return {string}
+             */
+            getCode: function() {
+                return 'omise_offline_tesco';
+            },
+
+            /**
+             * Initiate observable fields
+             *
+             * @return this
+             */
+            initObservable: function() {
+                this._super()
+                    .observe([
+                        'omiseOffsite'
+                    ]);
+
+                return this;
+            },
+
+            /**
+             * Get a checkout form data
+             *
+             * @return {Object}
+             */
+            getData: function() {
+                return {
+                    'method': this.item.method,
+                    'additional_data': {
+                        'offsite': this.omiseOffsite()
+                    }
+                };
+            },
+
+            /**
+             * Is method available to display
+             *
+             * @return {boolean}
+             */
+            isActive: function() {
+                return true;
+            },
+
+            /**
+             * Handle payment error
+             *
+             */
+            handlePaymentError: function(response) {
+                errorProcessor.process(response, self.messageContainer);
+                fullScreenLoader.stopLoader();
+                self.isPlaceOrderActionAllowed(true);
+            },
+
+            /**
+             * Hook the placeOrder function.
+             * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
+             *
+             * @return {boolean}
+             */
+            placeOrder: function(data, event) {
+                var self = this;
+
+                if (event) {
+                    event.preventDefault();
+                }
+
+                self.getPlaceOrderDeferredObject()
+                    .fail(
+                        function(response) {
+                            self.handlePaymentError(response);
+                        }
+                    ).done(
+                        function(response) {
+                            var self = this;
+
+                            var serviceUrl = urlBuilder.createUrl(
+                                '/orders/:order_id/omise-offsite',
+                                {
+                                    order_id: response
+                                }
+                            );
+
+                            storage.get(serviceUrl, false)
```

Comment:
@jacekstan just wonder, since Tesco Bill Payment doesn't require to redirect out to a 3rd-party page, do we need this whole `storage.get(serviceUrl, false).fail().done()` part?